http-client/src/http_client.rs

```diff
@@ -52,29 +49,28 @@ fn format_headers(h: &HeaderMap) -> HashMap<String, String> {
 
 fn build_request(
     client: &Client,
-    req: Request,
-) -> Result<RequestBuilder, Box<dyn std::error::Error>> {
-    let path = req.path.as_str();
+    req: RequestArgs,
+) -> Result<RequestBuilder, Box<dyn std::error::Error + Send + Sync>> {
     let mut r = match req.method.as_str() {
         "GET" => {
-            let r = client.get(format!("{}/{}", path, req.query_string).as_str());
+            let r = client.get(&req.url);
             Ok(r)
         }
         "POST" => {
-            let r = client.post(path).body(req.body);
+            let r = client.post(&req.url).body(req.body);
```

Comment:
This let should've been marked by clippy as superfluous. Same with the one on line 56.